### PR TITLE
[sda-auth] Make elixir endpoint optional

### PR DESCRIPTION
**Pull request by dbampalikis**
This PR makes the Elixir endpoint of the sda-auth optional. Specifically, if the variables `Elixir_id` or `Elixir_secret` are not set, the sda-auth is running only with the EGA option.

Also, in case neither the EGA nor the Elixir options are set, the service will return an error, letting the user know that it cannot start.

The PR is now deployed on the fega-test namespace

### CODE REVIEW

**Review by kusalananda — CHANGES_REQUESTED** [review]
Logic hiccup.

File: sda-auth/config.go [hunk 1]
```diff
@@ -120,6 +120,11 @@ func (c *Config) readConfig() error {
 
 	c.Cega = cega
 
+	// Check the either cega or Elixir config exists
+	if (elixir.ID == "" && elixir.Secret == "") && (cega.ID == "" && cega.Secret == "") {
```
[the remark this answers is not in the document]
Comment (dbampalikis) [reply sda-auth/config.go:124]: :sweat_smile: you are right, not sure what I was thinking :smile:

**Review by kusalananda — DISMISSED** [review]

File: sda-auth/main.go [hunk 2]
```diff
@@ -81,8 +81,12 @@ func (auth AuthHandler) getMain(ctx iris.Context) {
 // getLoginOptions returns the available login providers as JSON
 func (auth AuthHandler) getLoginOptions(ctx iris.Context) {
 
-	// Elixir is always available
-	response := []LoginOption{{Name: "Elixir", URL: "/elixir"}}
+	var response []LoginOption
+	if auth.Config.Elixir.ID != "" && auth.Config.Elixir.Secret != "" {
+		// Elixir is always available
```
Comment (pontus) [inline sda-auth/main.go:86]:
```suggestion
		// Elixir is configured, offer it
```

**Review by pontus — DISMISSED** [review]

**Comment by pontus** [comment]
Possibly add a test case verifying that lack of configuration results in failure?

**Review by kusalananda — DISMISSED** [review]

**Comment by dbampalikis** [comment]
> Possibly add a test case verifying that lack of configuration results in failure?

Makes sense, added a couple of cases for this configuration

**Review by pontus — APPROVED** [review]

**Review by kostas-kou — DISMISSED** [review]

**Review by kusalananda — DISMISSED** [review]
3rd time's the charm

**Review by kusalananda — APPROVED** [review]

File: sda-auth/README.md [hunk 3]
```diff
@@ -60,3 +60,7 @@ Using the provided Dockerfile, you may build a Docker image:
 ```bash
 docker build -t neicnordic/sda-auth:mytag <path-to-Dockerfile-folder>
 ```
+
+## Choosing provider login
+
+The `sda-auth` allows for two different type of login providers: `EGA` and `Elixir` (LS-AAI). It is possible, however, to run the `sda-auth` using only one of the parameters. In order to remove the `EGA` option, remove the `CEGA_ID` and `CEGA_SECRET` options for the configuration, while for removing the `Elixir` option, remove the `ELIXIR_ID` and `ELIXIR_SECRET` variables.
```
Comment (aaperis) [inline sda-auth/README.md:66]:
```suggestion
The `sda-auth` allows for two different type of login providers: `EGA` and `Elixir` (LS-AAI). It is possible, however, to run the `sda-auth` using only one of the providers. In order to remove the `EGA` option, remove the `CEGA_ID` and `CEGA_SECRET` options for the configuration, while for removing the `Elixir` option, remove the `ELIXIR_ID` and `ELIXIR_SECRET` variables.
```

**Review by aaperis — APPROVED** [review]
Looks good, minor typo :-)

**Review by aaperis — APPROVED** [review]

**Review by kusalananda — APPROVED** [review]

**Pull request merged**